lib/_stream_readable.js

```diff
@@ -593,9 +595,12 @@ function emitReadable_(stream) {
   debug('emitReadable_', state.destroyed, state.length, state.ended);
   if (!state.destroyed && (state.length || state.ended)) {
     stream.emit('readable');
-    state.emittedReadable = false;
   }
 
+  // Reset emittedReadable once it is safe to schedule another
+  // emitReadable.
+  state.emittedReadable = false;
```

Comment:
@mcollina: Please also observe this change. I'm pretty sure it's fine but just making sure I have a second opinion on that.